deploy/helm/sumologic/templates/_helpers.tpl

```diff
@@ -353,27 +353,157 @@ Generate metrics match configuration
 Example usage (as one line):
 
 {{ include "utils.metrics.match" (dict 
-  "Values" .Values 
-  "Match" "prometheus.metrics.kubelet" 
+  "Values" . 
+  "Tag" "prometheus.metrics.kubelet" 
   "Endpoint" "SUMO_ENDPOINT_METRICS" 
   "Storage" .Values.fluentd.buffer.filePaths.metrics.default
   "Id" sumologic.endpoint.metrics
 )}}
 */}}
 {{- define "utils.metrics.match" -}}
-<match {{ .Match }}>
+<match {{ .Tag }}>
+{{- if .Drop }}
+  @type null
+{{- else }}
   @type sumologic
   @id {{ .Id }}
-  endpoint "#{ENV['{{ .Endpoint }}']}"
-{{- .Values.fluentd.metrics.outputConf | nindent 2 }}
+  sumo_client {{ include "sumologic.sumo_client" .Context | quote }}
+  endpoint "#{ENV['{{ include "terraform.sources.endpoint" .Endpoint}}']}"
+{{- .Context.Values.fluentd.metrics.outputConf | nindent 2 }}
```

Comment:
I need to pass k8s context to have access to .Values, `.Context` in this template is simply `.` from kubernetes. 
`.` is override in `range` functions and templates, so there is need to pass it as a variable (or a dict value for templates, because they get only one argument). Here is flow (btw it covers both cases):
https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/8e58e3f3f1da3a704e7b2a70bd166c1cd9aa1012/deploy/helm/sumologic/conf/metrics/metrics.conf#L50
https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/8e58e3f3f1da3a704e7b2a70bd166c1cd9aa1012/deploy/helm/sumologic/conf/metrics/metrics.conf#L63